microcosm_flask/tests/test_namespaces.py

```diff
@@ -100,3 +101,24 @@ def search_foo():
     with graph.app.test_request_context():
         url = ns.href_for(Operation.Search)
         assert_that(url, is_(equal_to("http://localhost/api/foo")))
+
+
+def test_namespace_accepts_controller():
```

Comment:
Right now namespaces don't really do anything with the controller. It is just that I think it isn't entirely unreasonable for a namespace to declare who owns the implementation of routes for a given namespace.

We aren't doing this now, but we could use this to simplify the crud conventions to try to infer definitions from the namespace rather than explicitly registering them when configuring a controller, e.g.:

```
 def add_crud(graph, version):
     controller = graph.user_event_controller
      ns = Namespace(
          subject=UserEvent,
          controller=controller,
          version=version,
      )
      mappings = {
          Operation.Create: EndpointDefinition(
              func=transactional(controller.create),
             request_schema=NewUserEventSchema(),
             response_schema=UserEventSchema(),
         ),
         Operation.Delete: EndpointDefinition(
             func=transactional(controller.delete),
         ),
         Operation.Replace: EndpointDefinition(
             func=transactional(controller.replace),
             request_schema=NewUserEventSchema(),
             response_schema=UserEventSchema(),
         ),
         Operation.Retrieve: EndpointDefinition(
             func=controller.retrieve,
             response_schema=UserEventSchema(),
         ),
         Operation.Search: EndpointDefinition(
             func=controller.search,
             request_schema=UserEventSearchSchema(),
             response_schema=UserEventSchema(),
         ),
```

Could be:

```
 def add_crud(graph, version):
      controller = graph.user_event_controller
      ns = Namespace(
          subject=UserEvent,
          controller=controller,
          version=version,
      )
      mappings = {
          Operation.Create: EndpointDefinition(
             request_schema=NewUserEventSchema(),
             response_schema=UserEventSchema(),
         ),
         Operation.Delete: EndpointDefinition(
         ),
         Operation.Replace: EndpointDefinition(
             request_schema=NewUserEventSchema(),
             response_schema=UserEventSchema(),
         ),
         Operation.Retrieve: EndpointDefinition(
             response_schema=UserEventSchema(),
         ),
         Operation.Search: EndpointDefinition(
             request_schema=UserEventSearchSchema(),
             response_schema=UserEventSchema(),
         ),
```

For the sunny day case. Not sure how much it is worth, but pushing an instance of the controller to every definition seems redundant for something that should be constant per namespace.